graphdatascience/aura_api.py

```diff
@@ -91,7 +92,16 @@ def __init__(self, client_id: str, client_secret: str, tenant_id: Optional[str]
         self._logger = logging.getLogger()
         self._tenant_id = tenant_id if tenant_id else self._get_tenant_id()
 
-    def create_instance(self, name: str) -> InstanceCreateDetails:
+    @classmethod
+    def extract_id(cls, uri: str) -> str:
+        host = urlparse(uri).hostname
+
+        if not host:
+            raise RuntimeError(f"Could not parse the uri {uri}.")
```

Comment:
I like wrapping user-input values in backticks, especially if we failed to parse it. could contain odd whitespaces for instance.
```suggestion
            raise RuntimeError(f"Could not parse the uri `{uri}`.")
```